python-sdk/src/astro/files/base.py

```diff
@@ -50,6 +51,25 @@ def type(self) -> FileType:  # noqa: A003
             normalize_config=self.normalize_config,
         )
 
+    def to_json(self):
+        filetype = None if not self.filetype else self.filetype.value
+        return {
+            "class": "File",
+            "conn_id": self.conn_id,
+            "path": self.path,
+            "filetype": filetype,
+            "normalize_config": self.normalize_config,
+        }
+
+    @classmethod
+    def from_json(cls, serialized_object: dict):
+        return File(
+            conn_id=serialized_object["conn_id"],
+            path=serialized_object["path"],
+            filetype=constants.FileType(serialized_object["filetype"]),
+            normalize_config=serialized_object["normalize_config"],
+        )
```

Comment:
Our File and Table object are attrs classes, so you can do the following to serialize them natively.

```python
In [6]: from astro.files import File
In [7]: import attrs

In [8]: attrs.asdict(File("/path"))
Out[8]:
{'path': '/path',
 'conn_id': None,
 'filetype': None,
 'normalize_config': None,
 'uri': 'astro+file:/path',
 'extra': {}}


In [9]: from astro.sql.table import Table, Metadata

In [11]: attrs.asdict(Table("ta", conn_id="gcp", metadata=Metadata(schema="s")))
Out[11]:
{'_name': 'ta',
 'conn_id': 'gcp',
 'metadata': {'schema': 's', 'database': None},
 'columns': [],
 'temp': False,
 'uri': 'astro://gcp@?table=ta&schema=s',
 'extra': {}}
```

Reply:
@kaxil  I'm not sure this solution is worth it for two reason:
1. Some of the fields are not usable for deserialization (for example in File, the uri field will break instantiation). If we can't fix this, then it might be worth it for us to manually control what fields we serialize.
2. There doesn't seem to be an attrs.fromdict so deserialization would need to be handled by hand
Do you have any thoughts on this? It's highly possible I'm missing something here

Reply:
@kaxil another issue I found :

```
TypeError: TempTable.__init__() got an unexpected keyword argument '_name'
```

This is when I use the 

```
        table_dict = attrs.asdict(obj, recurse=True, filter=lambda a, v: a.init)
```

thing you showed me. I suppose we could filter for the _name field but it's starting to feel like this is a round hole square peg situation.